src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -354,7 +354,7 @@ private function parseDefinition($id, $service, $file, array $defaults)
                 }
 
                 if ('deprecated' === $key) {
-                    $alias->setDeprecated(true, $value);
+                    $alias->setDeprecated(true, \is_bool($value) ? null : $value);
```

Comment:
`true === $value ? null ? $value`?

Reply:
Actually, the same code for services doesn't have this logic, so this change might not be needed.

Reply:
Yes indeed, but if we have specified `deprecated: true` in the alias configuration, without this condition an error is thrown because we do not give a string containing `%alias_id%` but `true`...